.eslintrc

```diff
@@ -14,6 +16,7 @@
     "import/no-named-as-default": 2,
     "comma-dangle": 0,  // not sure why airbnb turned this on. gross!
     "indent": [2, 2, {"SwitchCase": 1}],
+    "no-case-declarations": 0,
```

Comment:
I solved it with a huge curly brace around the case statement like this:

case SAVE_SUCCESS:{
  const bla = zzz + xxx
  return { ...state, bla}
}
case whatever:
...

then you dont need to complicate the return object itself.
The rule accepts it like this because the declarations are isolated from other cases.

Sent from my Tricorder

> On 10.03.2016, at 22:46, utkuturunc <notifications@github.com> wrote:
> 
> In .eslintrc:
> 
> > @@ -14,6 +16,7 @@
> >      "import/no-named-as-default": 2,
> >      "comma-dangle": 0,  // not sure why airbnb turned this on. gross!
> >      "indent": [2, 2, {"SwitchCase": 1}],
> > +    "no-case-declarations": 0,
> It was kind of a personal choice. I like object-curly-spacing so I added spaces to all objects. Once you get used to this you cannot even look at non-spaced braces. 
> 
> About no-case-declarations, it makes reducer code more complicated. For example in the following code, definition of data raises an error. We will have to move that data declaration on top of the switch statement if we don't disable case-declarations property.
> 
> 
> case SAVE_SUCCESS:
> const data = [...state.data];
> data[action.result.id - 1] = action.result;
> return {
> ...state,
> data: data,
> editing: {
> ...state.editing,
> [action.id]: false
> },
> saveError: {
> ...state.saveError,
> [action.id]: null
> }
> };
> 
> —
> Reply to this email directly or view it on GitHub.
>